aks/storage_account/resources.tf

```diff
@@ -0,0 +1,82 @@
+locals {
+  storage_account_name = strcontains(var.environment, "review-") ? "${var.azure_resource_prefix}${var.service_short}${var.config_short}${replace(var.environment, "review-", "")}sa" : "${var.azure_resource_prefix}${var.service_short}${var.config_short}sa"
```

Comment:
not all services use review-nnnn for review apps, and strcontains is 1.5+
Maybe something like
storage_account_env = var.service_short != "rv" ?  var.config_short : replace(var.environment, "/^.*-/", "")
then
storage_account_name = "${var.azure_resource_prefix}${var.service_short}${local.storage_account_env}sa"